react/features/polls/components/PollDialog.js

```diff
@@ -0,0 +1,363 @@
+// @flow
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { DialogWithTabs, hideDialog } from '../../base/dialog';
+import PollCreateForm from './PollCreateForm';
+import VoteForm from './VoteForm';
+import PollResultsForm from './PollResultsForm';
+import { translate } from '../../base/i18n';
+import v1 from 'uuid/v1';
+import {
+    startPoll,
+    vote, endPoll
+} from '../actions';
+import { getUniquePollChoices } from '../functions';
+
+import { getLocalParticipant } from '../../base/participants';
+
+type Props = {
+
+    /**
+     * Redux dispatch method.
+     */
+    dispatch: Function,
+
+    /**
+     * True if there is an active poll session now.
+     */
+    isPollRunning: boolean,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function,
+
+    /**
+     * Current user id.
+     */
+    userID: string
+};
+
+type State = {
+
+    /**
+     * Available poll choices.
+     */
+    choices: Object,
+
+    /**
+     * New Poll Object
+     */
+    poll: Object,
+
+    /**
+     * Poll question.
+     */
+    question: Object
+};
+
+/**
+ * Polls main dialog view component.
+ */
+class PollDialog extends Component<Props, State> {
+
+    /**
+     * Constructor.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._closeDialog = this._closeDialog.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+        this._addChoice = this._addChoice.bind(this);
+        this._removeChoice = this._removeChoice.bind(this);
+        this._onQuestionTextChange = this._onQuestionTextChange.bind(this);
+        this._createNewPoll = this._createNewPoll.bind(this);
+        this._choiceTextChange = this._choiceTextChange.bind(this);
+        this._vote = this._vote.bind(this);
+        this._endPoll = this._endPoll.bind(this);
+        this._clearState = this._clearState.bind(this);
+
+        const initialState = this._clearState();
+
+        this.state = {
+            ...initialState
+        };
+    }
+
+    _closeDialog: () => void;
+
+    /**
+     * Callback invoked to close the dialog without saving changes.
+     *
+     * @private
+     * @returns {void}
+     */
+    _closeDialog() {
+        this.props.dispatch(hideDialog());
```

Comment:
I'm surprised you have to do this for DialogWithTabs. Normally with web dialog you return true to confirm closing.

Reply:
I tried to replace this statement with `return true;` but didn't work.
Looking up other usages of `DialogWithTabs` this looks like the way to close it.
For example:
react/features/settings/components/web/SettingsDialog.js